www/components/messagebox.tsx

```diff
@@ -1,28 +1,74 @@
+import { useCallback, useState } from "react";
 import Image from "next/image";
 import icon from "@/public/bloomicon.jpg";
 import usericon from "@/public/usericon.svg";
 import Skeleton from "react-loading-skeleton";
+import { FaLightbulb, FaThumbsDown, FaThumbsUp } from "react-icons/fa";
+import { API } from "@/utils/api";
+import useSWR, { mutate } from "swr";
 
-interface MessageBoxRegularProps {
-  children: React.ReactNode;
+interface MessageBoxProps {
   isUser?: boolean;
-  loading?: false;
+  messageId?: string;
+  conversationId?: string;
+  api?: API;
+  text: string;
+  loading?: boolean;
+  isThoughtsOpen?: boolean;
+  setIsThoughtsOpen: (isOpen: boolean) => void;
+  setThought: (thought: string) => void;
 }
 
-interface MessageBoxLoadingProps {
-  children?: React.ReactNode;
-  isUser?: boolean;
-  loading: true;
-}
-
-// merge the two types
-type MessageBoxProps = MessageBoxRegularProps | MessageBoxLoadingProps;
+const fetchThought = async (key: string) => {
+  const [_, conversationId, messageId, userId] = key.split(":");
+  const api = new API({ url: process.env.NEXT_PUBLIC_API_URL!, userId });
+  return api.getThoughtById(conversationId, messageId);
+};
 
 export default function MessageBox({
-  children,
   isUser,
-  loading,
+  messageId,
+  api,
```

Comment:
Instead of taking an instance of an API as a property it might be more efficient to generate an instance when needed. Save on initial rendering time.